modules/islandora_scopus/includes/utilities.inc

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @file
+ * Utility functions for Scopus
+ */
+
+/**
+ * Find the SCOPUS ID using an other id like DOI, PMID, etc.
+ *
+ * @param string $type
+ *   The type of id to use (doi).
+ * @param string $code
+ *   The value of the code to search for.
+ *
+ * @return mixed
+ *   Return the Scopus ID or FALSE.
+ */
+function islandora_scopus_get_id($type, $code) {
+  $api_key = variable_get('islandora_scopus_api_key', '');
+
+  switch (strtolower($type)) {
+    case "doi":
+      $query = "DOI%28" . urlencode($code) . "%29";
```

Comment:
It appears that the `drupal_http_request` passes the data in the message body and Scopus disregards that message body as this is a `GET` request. So this
```
$doi_string = format_string('DOI(!doi)', array('!doi' => $code));
$query = http_build_query(array('query' => $doi_string));
$query_url = "http://api.elsevier.com/content/search/scopus";
$headers = array('Accept' => 'application/json', 'X-ELS-APIKey' => $api_key);
$response = drupal_http_request($query_url, array('headers' => $headers, 'method' => 'GET', 'data' => $query));
```
fails, while this
```
$query = http_build_query(array('query' => $doi_string));
$query_url = "http://api.elsevier.com/content/search/scopus?{$query}";
$headers = array('Accept' => 'application/json', 'X-ELS-APIKey' => $api_key);
$response = drupal_http_request($query_url, array('headers' => $headers, 'method' => 'GET'));
```
works. This is up to Scopus's implementation but could be because a `GET` request (IMHO) should not use the message body to send information vital to deriving the Request URI.